e2e/cluster.go

```diff
@@ -231,6 +231,20 @@ func getOrCreateCluster(ctx context.Context, cluster *armcontainerservice.Manage
 		return nil, fmt.Errorf("failed to get cluster %q: %w", *cluster.Name, err)
 	}
 	logf(ctx, "cluster %s already exists in rg %s", *cluster.Name, resourceGroupName)
+
+	// ensure MC_rg as well
+	nodeRGExists, err := isExistingResourceGroup(ctx, *existingCluster.Properties.NodeResourceGroup)
```

Comment:
line 250 has the exact same check. Can you see if it's enough to add the deleteCluster call there?